samples/xaudio/main.c

```diff
@@ -0,0 +1,79 @@
+#include <hal/debug.h>
+#include <hal/video.h>
+#include <hal/xbox.h>
+#include <windows.h>
+#include <string.h>
+#include <hal/audio.h>
+#include <xboxkrnl/xboxkrnl.h>
+#include <assert.h>
+
+#define MIN(x,y) ((x)<(y)?(x):(y))
+#include "nxdk_wav.h"
+
+unsigned int callback_count;
+
+unsigned char *voice_data;
+unsigned int voice_len;
+unsigned int voice_pos;
+
+#define NUM_BUFFERS 2
+const unsigned short buffer_size = 48*1024;
+unsigned char *buffers[NUM_BUFFERS];
+unsigned int current_buf;
+
+static void provide_samples_callback(void *pac97Device, void *data)
```

Comment:
How is this called? I could imagine that this doesn't allow float as it is an interrupt (do DPCs allow float?)

Reply:
It's called from a DPC. No floats are used here, but if someone uses floats here they will need to save and restore the FPU state. I added a comment to warn people.